src/python_testing/TC_RVCCLEANM_1_2.py

```diff
@@ -55,64 +69,74 @@ async def test_TC_RVCCLEANM_1_2(self):
+            # Verify that each ModeOptionsStruct entry has a unique Mode field value
             for m in supported_modes:
-                if m.mode in modes:
+                if m.mode in self.supported_modes_dut:
                     asserts.fail("SupportedModes must have unique mode values!")
                 else:
-                    modes.append(m.mode)
+                    self.supported_modes_dut.append(m.mode)
 
+            # Verify that each ModeOptionsStruct entry has a unique Label field value
             labels = []
             for m in supported_modes:
                 if m.label in labels:
                     asserts.fail("SupportedModes must have unique mode label values!")
                 else:
                     labels.append(m.label)
 
-            # common mode tags
-            commonTags = {0x0: 'Auto',
-                          0x1: 'Quick',
-                          0x2: 'Quiet',
-                          0x3: 'LowNoise',
-                          0x4: 'LowEnergy',
-                          0x5: 'Vacation',
-                          0x6: 'Min',
-                          0x7: 'Max',
-                          0x8: 'Night',
-                          0x9: 'Day'}
-
-            cleanTags = [tag.value for tag in Clusters.RvcCleanMode.Enums.ModeTag
-                         if tag is not Clusters.RvcCleanMode.Enums.ModeTag.kUnknownEnumValue]
+            # Verify that each ModeOptionsStruct entry's ModeTags field has:
+            for m in supported_modes:
+                # * at least one entry
+                if len(m.modeTags) == 0:
+                    asserts.fail("SupportedModes must have at least one mode tag!")
 
+                at_least_one_common_or_clean_tag = False
+                for t in m.modeTags:
+                    # * the values of the Value fields that are not larger than 16 bits
+                    if t.value > 0xFFFF or t.value < 0:
+                        asserts.fail("Mode tage values must not be lager tha 16 bits!")
+
+                    # * for each Value field: {isCommonOrDerivedOrMfgTagsVal}
+                    is_mfg = (0x8000 <= t.value <= 0xBFFF)
+                    if (t.value not in self.commonTags and
+                            t.value not in self.cleanTags and
+                            not is_mfg):
+                        asserts.fail("Mode tag value is not a common tag, clean tag or vendor tag!")
+
+                    # * for at least one Value field: {isCommonOrDerivedTagsVal}
+                    if not is_mfg:
+                        at_least_one_common_or_clean_tag = True
+
+                if not at_least_one_common_or_clean_tag:
+                    asserts.fail("At least one mode tag must be a common tag or clean tag!")
+
+            # Verify that at least one ModeOptionsStruct entry includes either the
+            # Vacuum(0x4001) mode tag or the Mop(0x4002)mode tag in the ModeTags field
+            has_vacuum_or_mop_mode_tag = False
```

Comment:
could you do 
vacuum_mop = [Clusters.RvcCleanMode.Enums.ModeTag.kVacuum, Clusters.RvcCleanMode.Enums.ModeTag.kMop]
has_vacuum_or_mop_mode_tag = any(tag in vacuum_mop for tag in supported_modes)

Reply:
This has been simplified to 
```
    # Verify that at least one ModeOptionsStruct entry includes either the
    # Vacuum(0x4001) mode tag or the Mop(0x4002)mode tag in the ModeTags field
    vacuum_mop_tags = [Clusters.RvcCleanMode.Enums.ModeTag.kVacuum, Clusters.RvcCleanMode.Enums.ModeTag.kMop]
    has_vacuum_or_mop_mode_tag = False
    for m in supported_modes:
        has_vacuum_or_mop_mode_tag = any(t.value in vacuum_mop_tags for t in m.modeTags)
        if has_vacuum_or_mop_mode_tag:
            break
```